src/datachain/data_storage/metastore.py

```diff
@@ -107,13 +113,95 @@ def cleanup_for_tests(self) -> None:
         """Cleanup for tests."""
 
     #
-    # Datasets
+    # Namespaces
+    #
+
+    @property
+    @abstractmethod
+    def default_namespace_name(self):
+        """Gets default namespace name"""
+
+    @abstractmethod
+    def create_namespace(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        uuid: Optional[str] = None,
+        ignore_if_exists: bool = True,
+        **kwargs,
+    ) -> Namespace:
+        """Creates new namespace"""
+
+    @abstractmethod
+    def get_namespace(self, name: str, conn=None) -> Namespace:
+        """Gets a single namespace by name"""
+
+    @abstractmethod
+    def list_namespaces(self, conn=None) -> list[Namespace]:
+        """Gets a list of all namespaces"""
+
+    @property
+    @abstractmethod
+    def is_studio(self) -> bool:
```

Comment:
do we use it enywhere beside allow_create_projects ?

I would try to remove it (have less studio specific code - just make metastore implement decide ... if it is not implemented let those methods raise?)